### ISSUE TITLE
Gestion des utilisateurs SP vs utilisateurs Hypothes.is

### ISSUE THREAD

**Issue by lakonis** [timoguic/sp_hub#15]
* Explorer l'api hypothes.is et une intégration fine pour faire correspondre les users SP avec les users hypothes.is. 
* à discuter : il serait préférable que les annotations "tombent" dans un groupe d'annotation dédié auquel on rattacherait tout nouvel utilisateur de SP via le compte hypothes.is qu'on créerait pour lui, et auquel on inviterait les utilisateurs d'hypothes.is non authentifié à SP. Cette invitation pourrait donner lieu à une création de compte sur SP.
  Le problème de ce groupe d'annotation est que les utilisateurs non-authentifiés risquent de ne pas pouvoir visualiser les annotations (et les discussions). 
  * [ ] Vérifier si l'API Hypths le permet.

**Comment by timoguic** [comment]
https://h.readthedocs.io/en/latest/api/#operation/createUser

Je ne connais pas très bien hypothes.is, mais la doc référence la possibilité pour "certains utilisateurs" de créer des utilisateurs "within their authority". Il me semblait avoir lu qu'il fallait un compte spécial, qui peut se rapprocher d'eux pour avoir des infos?

**Comment by lakonis** [comment]
Je m'en occupe. Liste de questions à affiner : 
  - [ ] gestion de groupes SP,
  - [ ] quid des personnes ayant déjà un compte hypothes.is
  - [ ] création d'un user sp_hub => création d'un user hypothes.is
  - [ ] visibilité publique des annotations d'un groupe
  - [ ] ...?

**Comment by timoguic** [comment]
> quid des personnes ayant déjà un compte hypothes.is

Effectivement. Il vaut peut être mieux prévoir une création manuelle du compte hypothesis (éventuellement assistée par la plateforme). L'utilisateur peut ensuite renseigner son ID hypothesis dans son profil (idem pour l'ORCID, non?). À voir suivant leur retour.

**Comment by lakonis** [comment]
ORCID fournit un service web permettant de requêter directement leur api. Il faut prévoir de l'implémenter pour les "auteurs" dont on sait déjà que les noms seront valides. Par contre pour les utilisateurs de la communauté, il faut leur laisser la possibilité de créer un compte sous pseudo, et dans ce cas pas d'alignement sur ORCID.

Pour l'ID hypothes.is, il y a le cas limite de l'auteur à qui on a créé par défaut un user SP_hub et qui souhaite y associer son propre id hypothes.is.. Dans ce cas, doit on maintenir 2 comptes hypothes.is (dans le cas où l'auteur aurait utilisé le compte hypothes.is par défaut) avant d'associer un deuxième compte ? complexe..

**Comment by timoguic** [comment]
D'où l'idée de simplement demander à l'utilisateur de rentrer son compte hypothesis à la main... Quitte à ce que l'éditeur le fasse. Sinon ça va être compliqué et pas forcément plus efficace.

**Comment by lakonis** [comment]
Entendu, ça me va. Et s'il n'en a pas, un bouton pour en créer un ? cad la plateforme s'occupe de créer un user hypothes.is et de l'associer au groupe d'annotation SP (si on choisit cette solution).

**Comment by lakonis** [comment]
Pour info : https://groups.google.com/d/msg/hypothesis-forum/OuBmfgpuw2k/zQY_QiNGAQAJ
